lib/hammer_cli_foreman/table_preference.rb

```diff
@@ -0,0 +1,48 @@
+module HammerCLIForeman
+  class TablePreference < HammerCLIForeman::Command
+    resource :table_preferences
+    command_name "table-preference"
+
+    class ListCommand < HammerCLIForeman::ListCommand
+      output do
+        field :id, _("Id")
+        field :name, _("Name")
+        field :columns, _("Columns"), Fields::List
+      end
+
+      build_options
+    end
+
+    class CreateCommand < HammerCLIForeman::CreateCommand
+      success_message _('Table preference created.')
+      failure_message _('Could not create table preference')
+
+      build_options
+    end
+
+    class UpdateCommand < HammerCLIForeman::UpdateCommand
+      success_message _('Table preference updated.')
+      failure_message _('Could not update table preference')
+
```

Comment:
Sorry, I forgot to mention that we need to remove automatically added `--new-name` option to not confuse the users. You can do it by removing searchables for this resource (we don't need them anyway). Just add a new item similarly to this: https://github.com/theforeman/hammer-cli-foreman/blob/master/lib/hammer_cli_foreman/id_resolver.rb#L69

```ruby
      :compute_attribute => [],
      :table_preference => []
```

Reply:
Thanks for suggestion, now it should be good